vnext/Microsoft.ReactNative.Cxx/JSValueReader-WinRT.h

```diff
@@ -0,0 +1,203 @@
+            {L"SystemAccentColorLight1", {nullptr}},
+            {L"SystemAccentColorLight2", {nullptr}},
+            {L"SystemAccentColorLight3", {nullptr}},
+            {L"SystemAccentColorDark1", {nullptr}},
+            {L"SystemAccentColorDark2", {nullptr}},
+            {L"SystemAccentColorDark3", {nullptr}},
+            {L"SystemListAccentLowColor", {nullptr}},
+            {L"SystemListAccentMediumColor", {nullptr}},
+            {L"SystemListAccentHighColor", {nullptr}}};
+
+        if (accentColorMap.find(resourceName) != accentColorMap.end()) {
+          if (auto brush = accentColorMap.at(resourceName).get()) {
+            value = brush;
+            return;
+          }
+
+          winrt::hstring xamlString =
+              L"<ResourceDictionary"
+              L"    xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'"
+              L"    xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'>"
+              L"  <SolidColorBrush"
+              L"      x:Key='" +
+              resourceName +
+              L"'"
+              L"      Color='{ThemeResource " +
+              resourceName +
+              "}' />"
+              L"</ResourceDictionary>";
+
+          auto dictionary{winrt::unbox_value<xaml::ResourceDictionary>(xaml::Markup::XamlReader::Load(xamlString))};
+
+          auto brush{
+              winrt::unbox_value<xaml::Media::SolidColorBrush>(dictionary.Lookup(winrt::box_value(resourceName)))};
+
+          accentColorMap[resourceName] = winrt::make_weak(brush);
+
+          value = brush;
+          return;
+        }
+
+        winrt::IInspectable resource{xaml::Application::Current().Resources().Lookup(winrt::box_value(resourceName))};
+
+        value = winrt::unbox_value<xaml::Media::SolidColorBrush>(resource);
+        return;
+      }
+    }
+    return;
+  }
+
+  ui::Color color;
+  ReadValue(reader, color);
+
+  thread_local static std::map<ui::Color, winrt::weak_ref<xaml::Media::SolidColorBrush>, ColorComp>
+      solidColorBrushCache;
+
+  if (solidColorBrushCache.count(color) != 0) {
+    if (auto brush = solidColorBrushCache[color].get()) {
+      value = brush;
+      return;
+    }
+  }
+
+  value = xaml::Media::SolidColorBrush(color);
+  solidColorBrushCache[color] = winrt::make_weak(value);
+}
+
+inline void ReadValue(IJSValueReader const &reader, /*out*/ xaml::VerticalAlignment &value) {
+  value = xaml::VerticalAlignment::Stretch;
+  if (reader.ValueType() == JSValueType::String) {
+    const auto valueString = reader.GetString();
+    if (valueString == L"bottom")
+      value = xaml::VerticalAlignment::Bottom;
+    else if (valueString == L"center")
+      value = xaml::VerticalAlignment::Center;
+    else if (valueString == L"stretch")
+      value = xaml::VerticalAlignment::Stretch;
+    else if (valueString == L"top")
+      value = xaml::VerticalAlignment::Top;
+    else {
+      // ASSERT: Invalid value for VerticalAlignment. Shouldn't get this far.
+      assert(false);
```

Comment:
>assert(false); [](start = 6, length = 14)

Consider adding a comment string for the assert. E.g. `assert(false && "Invalid value for VerticalAlignment. Shouldn't get this far.")`
The assert is a macro and its body is shown to the user when it fails.